src/schemes/fluid/entropically_damped_sph/system.jl

```
@@ -306,6 +306,10 @@ function update_final!(system::EntropicallyDampedSPHSystem, v, u, v_ode, u_ode,
     compute_curvature!(system, surface_tension, v, u, v_ode, u_ode, semi, t)
     compute_stress_tensors!(system, surface_tension, v, u, v_ode, u_ode, semi, t)
     update_average_pressure!(system, system.transport_velocity, v_ode, u_ode, semi)
+
+    # checks for callback dependent models
+    check_tvf_configuration(system, system.transport_velocity, v, u, v_ode, u_ode, semi, t;
```

Why is this the very last step of `update_final!`? Wouldn't it make more sense to make this the very first step of `update_quantities!`?

---

There is a comment in boundary/system.jl. But I don't remember why
```
# This update depends on the computed quantities of the fluid system and therefore
# has to be in `update_final!` after `update_quantities!`.
```

---

I don't remember why...

---

and now???

---

Idk. I think it makes more sense to do that at the beginning to make it fail faster for large problems.

---

But I also don't really care.